src/routes/authentication.js

```diff
@@ -137,6 +142,34 @@ async function authenticationCallback(context, options, request, response, next)
   }
 }
 
+/**
+ * @param {import('../context/init').Context} context
+ * @param {import('express').Request} request
+ * @param {import('express').Response} response
+ */
+async function logout(context, request, response) {
+  const cookies = request.headers.cookie;
+  const isLocalhost = request.headers.host.match(REGEX_LOCALHOST);
+
+  if (isLocalhost) {
+    response.status(204).send();
+  } else if (cookies) {
+    const match = cookies.match(REGEX_COOKIE_SESSION_TOKEN);
```

Comment:
This regex and this logic is defined at 2 places in the code. I suggested you to refactor the current code to extract this logic into a function that will be used in both places.

In addition to that, I think you can only keep ONE if statement in this function. You can test if the cookie is present. If it is, then you delete it, if it is not, then you do nothing.

Here you have a lot of identical code blocks for cases that can be easily merged.